src/Development/IDE/Core/Rules.hs

```diff
@@ -334,11 +334,21 @@ instance Show GhcSessionFun where show _ = "GhcSessionFun"
 instance NFData GhcSessionFun where rnf !_ = ()
 
 
+-- Rule type for caching GHC sessions.
```

Comment:
Given that this rule is only used in `Main.hs`, I think we should move it there (or some helper module for the executable). The `ghcide` library explicitly abstracts over how you setup the environment so that consumers like `daml` can plug in something cmpletely custom.

Reply:
I initially added this rule to `Main.hs` and realized that it required adding several new dependencies to the executable component: `hashable`, `deepseq`, and `binary`. I can move this back to `Main.hs` if that makes things cleaner for `daml`. Just let me know.

Reply:
Moved to `Main.hs`.